contracts/flashloan/liquidity.sol

```diff
@@ -486,34 +497,39 @@ contract DydxFlashloaner is Resolver, ICallee, DydxFlashloanBase, DSMath {
         Account.Info[] memory accountInfos = new Account.Info[](1);
         accountInfos[0] = _getAccountInfo();
 
-        uint[] memory iniBals = new uint[](_length);
-        uint[] memory finBals = new uint[](_length);
-        uint[] memory _feeAmts = new uint[](_length);
+        FlashLoanData memory flashloanData;
+
+        flashloanData._iniBals = new uint[](_length);
+        flashloanData._finBals = new uint[](_length);
+        flashloanData._feeAmts = new uint[](_length);
+        flashloanData._tokenDecimals = new uint[](_length);
         for (uint i = 0; i < _length; i++) {
-            iniBals[i] = _tokenContracts[i].balanceOf(address(this));
+            flashloanData._iniBals[i] = _tokenContracts[i].balanceOf(address(this));
+            flashloanData._tokenDecimals[i] = TokenInterface(address(_tokenContracts[i])).decimals();
```

Comment:
What in the case of ETH? As it's not a token.

Reply:
It's using WETH as token address